components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.clazz.ui/src/main/resources/web/class-mediator/update-mediator.jsp

```diff
@@ -26,17 +33,43 @@
         throw new RuntimeException("Unable to edit the mediator");
     }
     ClassMediator classMediator = (ClassMediator) mediator;
+
+    XPathFactory xPathFactory = XPathFactory.getInstance();
     String propertyCountParameter = request.getParameter("propertyCount");
     classMediator.getProperties().clear(); // to avoid duplications
     if (propertyCountParameter != null && !"".equals(propertyCountParameter)) {
         int propertyCount = 0;
-        propertyCount = Integer.parseInt(propertyCountParameter.trim());
-        for(int i = 0;i < propertyCount ;i++){
-            String propertyName = request.getParameter("propertyName" + i);
-            String propertyValue = request.getParameter("propertyValue" + i);
-            if(propertyName!=null && propertyValue !=null){
-                classMediator.addProperty(propertyName,propertyValue);
+        try {
+            propertyCount = Integer.parseInt(propertyCountParameter.trim());
+            for (int i = 0; i <= propertyCount; i++) {
+                String name = request.getParameter("propertyName" + i);
+                if (name != null && !"".equals(name)) {
```

Comment:
Shall we use isEmpty() check instead of ""?

Reply:
+1